modules/index/src/main/java/org/apache/ignite/internal/index/IndexBuilder.java

```diff
@@ -100,10 +111,14 @@ void stop() {
      */
     void startIndexBuild(TableIndexView tableIndexView, TableImpl table) {
         for (int partitionId = 0; partitionId < table.internalTable().partitions(); partitionId++) {
+            // Let's check if there is a node in the assignments for the partition.
+            if (!replicaManager.startedGroups().contains(new TablePartitionId(table.tableId(), partitionId))) {
```

Comment:
Why do we need to iterate over all partitions and check if `startedGroups` contains a particular one? Can we simply iterate over `startedGroups` and get partitions from there?

Reply:
The checking based on started replica does not look like a permanent solution. Because the replica is a consequence of assignments.
If it is a temporary, that required TODO here.

Reply:
I used replicas on the advice of @sanpwc.

Reply:
That's definitely not what I've meant. What about having build index logic inside replica itself? The basic question here is whether we have any index related logic if there's no replica locally? 
If not, we should neither check assignments nor replicas but rather perform index related logic inside replica. Of course it's still possible to used IndexBuilder as an abstraction that will cover corresponding index related logic.